src/components/core/charts/LineChart/utils.ts

```diff
@@ -90,6 +91,64 @@ export const calculateVisibleYRange = <
   };
 };
 
+/**
+ * Calculates the visible y-axis range for the chart.
+ * - Min value: min(0, dataMinValue) - starts at 0 for positive data, or the negative minimum
+ * - Max value: dataMaxValue * 1.2 when positive (20% headroom), or 0 when all values are negative
+ * - When all values are identical, ensures a reasonable range (min -1 to 0 for negatives, 0 to 1 for small positives)
+ */
+export const calculateVisibleYRange = <
+  V extends number | string,
+  T extends ChartDataPoint<V>,
+>(
+  data: T[],
+) => {
+  const values = data.map((d) => Number(d.value)).filter((v) => !isNaN(v));
+
+  if (values.length === 0) {
+    return {
+      minValue: 0,
+      maxValue: 1,
+      minValueWithOffset: 0,
+      maxValueWithOffset: 1,
+      isSymmetricRange: false,
+      isNegative: false,
+    };
+  }
+
+  const dataMinValue = Math.min(...values);
+  const dataMaxValue = Math.max(...values);
+
+  const isNegative = dataMinValue < 0;
+  const isSame = dataMinValue === dataMaxValue;
```

Comment:
Do you need an epsilon here?


```suggestion
  const isSame = Math.abs(dataMinValue - dataMaxValue) < Number.EPSILON // or even a larger value like 0.1
```